components/home/Community.vue

```diff
@@ -57,76 +57,154 @@
-            :options="[
-              'Asia',
-              'North America',
-              'Europe',
-              'Australia',
-              'South America',
-              'Africa',
-            ]"
-            placeholder="Select Location"
-            style="margin-top: 0.625rem"
-          />
-          <v-label value="email" strong style="margin-top: 3rem" />
-          <v-text-field
-            v-model="email"
-            placeholder="Enter here"
-            style="margin-top: 0.625rem"
-          />
-          <v-button
-            label="submit"
-            label-transform="uppercase"
-            style="margin-top: 3rem; width: 12rem"
-          />
+            <v-label value="email" strong style="margin-top: 3rem" />
+            <v-text-field
+              v-model="email"
+              placeholder="Enter here"
+              style="margin-top: 0.625rem"
+            />
+            <v-button
+              label="submit"
+              label-transform="uppercase"
+              style="margin-top: 3rem; width: 12rem"
+              :action="submitProvider"
+            />
+          </v-stack>
+          <v-text class="subscription-message" :class="{ success }">
+            {{ message }}
+          </v-text>
         </v-stack>
-        <v-image path="images/provide-illustration.svg" class="illustration" />
+        <v-image path="images/provide-illustration.png" class="illustration" />
       </v-stack>
     </v-container>
   </section>
 </template>
 
 <script>
+import { subscribe } from '~/services/mailchimp'
+
 export default {
   name: 'Community',
   data() {
     return {
-      providerType: 'Storage Provider',
+      providerType: '',
       capacity: {
         value: 100,
         unit: 'GB',
       },
       location: '',
       email: '',
+      success: true,
+      message: '',
     }
   },
+  methods: {
+    async submitProvider() {
+      if (this.email.trim() && this.providerType && this.location) {
+        this.success = true
+        this.message = 'Submitting...'
+        try {
+          const data = {
+            email: this.email,
+            groups: ['Provider', 'Newsletter'],
+            providerType: this.providerType,
+            location: this.location,
+          }
+          if (this.providerType === 'Storage Provider') {
+            data.providerType = 'Storage'
+            data.storage = {
+              capacity: this.capacity.value,
+              unit: this.capacity.unit,
+            }
+          }
+          await subscribe(data)
+          this.message = 'Thank you for subscribing!'
+        } catch (e) {
+          this.success = false
+          if (/subscribed/.test(e)) {
+            this.message = 'Already Subscribed'
+          } else if (/0 - /.test(e)) {
+            this.message = 'Invalid email'
+          }
+        }
+      } else {
+        this.success = false
+        this.message = 'Enter all details to continue'
+      }
+    },
+  },
```

Comment:
This code will submit the details to mailchimp. Currently it is repetitive. Need a way to avoid repetitiveness